shared/src/crypto.rs

```diff
@@ -287,12 +288,24 @@ impl CertificateCache {
     pub async fn update_certificates_mut(&mut self) -> Result<CertificateCacheUpdate, SamplyBeamError> {
         debug!("Updating certificates via network ...");
         let certificate_list = CERT_GETTER.get().unwrap().certificate_list_via_network().await?;
-        let new_certificate_serials: Vec<&String> = {
-            certificate_list
-                .iter()
-                .filter(|serial| !self.serial_to_x509.contains_key(*serial))
-                .collect()
+        let certificate_revocation_list = CERT_GETTER.get().unwrap().get_crl().await?;
+        // Check if any of the certs in the cache have been revoked
+        let mut revoked_certs = 0;
+        if let Some(ref crl) = certificate_revocation_list {
+            self.serial_to_x509.values_mut().for_each(|cert_entry| {
+                if let CertificateCacheEntry::Valid(ref cert) = cert_entry {
+                    if is_revoked(crl.get_by_cert(cert)) {
+                        *cert_entry = CertificateCacheEntry::Invalid(CertificateInvalidReason::Revoked);
+                        revoked_certs += 1;
+                    }
+                }
+            })
         };
+        debug!("Revoked {revoked_certs} certificates from cache.");
```

Comment:
This should be in its own function and, for example, be called whenever the CRL changes. For example, in a later iteration the beam-proxy might be told via the control channel to please fetch the new CRL.